mcp/CMakeLists.txt

```diff
@@ -26,13 +26,38 @@ function(define_file_basename_for_sources targetname)
 endfunction()
 
 # Have each target run its own style check on recompile.
+#
+# Note
+# ----
+# Optional patterns (regular expressions) can be provided to specify files
+# for which linting should be skipped (e.g. files from other projects which
+# were included as-is to the source base).
 function(check_style targetname)
-    # find all sources (headers and source files!) that contribute to this target
-    get_target_property(check_files "${targetname}" SOURCES)
-    # remove some big ones that shouldn't change
-    list(REMOVE_ITEM check_files ${PROJECT_SOURCE_DIR}/sensors/geomag2015.c ${PROJECT_SOURCE_DIR}/sensors/../include/geomag2015.h ${PROJECT_SOURCE_DIR}/sensors/../include/EGM9615.h)
```

Comment:
There is a sneaky header file here that describes the shape of the Earth - a model called EGM 96. It's no longer being excluded in a hard-coded manner, and the linter is breaking the flight software build. So you'll have to go into any targets that include it and add it to a regex like you have to the geomag model. You could also keep excluding it like we had been before, since EGM96 updates are less frequent and less impactful.

Reply:
Right, I'm working on it. I already modified `check_style()` functions to accept regex patterns which allow one to use regex patterns to exclude files for which style checking should be skipped.